source/reporters/ut_junit_reporter.tpb

```diff
@@ -35,55 +35,62 @@ create or replace type body ut_junit_reporter is
     end;
 
     procedure print_test_elements(a_test ut_test) is
-      l_lines ut_varchar2_list;
+      l_results ut_varchar2_rows := ut_varchar2_rows();
+      l_lines   ut_varchar2_list;
       l_output clob;
     begin
-      self.print_text('<testcase classname="' || dbms_xmlgen.convert(get_path(a_test.path, a_test.name)) || '"' || ' assertions="' ||
-                      nvl(a_test.all_expectations.count,0) || self.get_common_item_attributes(a_test) || case when
-                      a_test.result != ut_utils.gc_success then
-                      ' status="' || ut_utils.test_result_to_char(a_test.result) || '"' end || '>');
+      ut_utils.append_to_list(
+        l_results,
+        '<testcase classname="' || dbms_xmlgen.convert(get_path(a_test.path, a_test.name)) || '"' || ' assertions="'
+        || nvl(a_test.all_expectations.count,0) || self.get_common_item_attributes(a_test)
+        || case when a_test.result != ut_utils.gc_success then
+                ' status="' || ut_utils.test_result_to_char(a_test.result) || '"' end || '>'
+      );
       if a_test.result = ut_utils.gc_disabled then
-        self.print_text('<skipped/>');
+        ut_utils.append_to_list( l_results, '<skipped/>' );
       end if;
       if a_test.result = ut_utils.gc_error then
-        self.print_text('<error>');
-        self.print_text(c_cddata_tag_start);
-        self.print_clob(ut_utils.table_to_clob(a_test.get_error_stack_traces()));
-        self.print_text(c_cddata_tag_end);
-        self.print_text('</error>');
+        ut_utils.append_to_list( l_results, '<error>');
+        ut_utils.append_to_list( l_results, c_cddata_tag_start);
+        ut_utils.append_to_list( l_results, ut_utils.convert_collection(a_test.get_error_stack_traces()) );
+        ut_utils.append_to_list( l_results, c_cddata_tag_end);
+        ut_utils.append_to_list( l_results, '</error>');
       elsif a_test.result > ut_utils.gc_success then
-        self.print_text('<failure>');
+        ut_utils.append_to_list( l_results, '<failure>');
         for i in 1 .. a_test.failed_expectations.count loop
           
           l_lines := a_test.failed_expectations(i).get_result_lines();
           
           for j in 1 .. l_lines.count loop
-            self.print_text(dbms_xmlgen.convert(l_lines(j)));
+            ut_utils.append_to_list( l_results, dbms_xmlgen.convert(l_lines(j)) );
           end loop;
-          self.print_text(dbms_xmlgen.convert(a_test.failed_expectations(i).caller_info));
+          ut_utils.append_to_list( l_results, dbms_xmlgen.convert(a_test.failed_expectations(i).caller_info) );
         end loop;
-        self.print_text('</failure>');
+        ut_utils.append_to_list( l_results, '</failure>');
       end if;
       -- TODO - decide if we need/want to use the <system-err/> tag too
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Complete the task associated to this "TODO" comment. ([plsql:S1135](https://sonarcloud.io/organizations/utplsql/rules#rule_key=plsql%3AS1135))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=utPLSQL&pullRequest=783&issues=AWcfCxK8d6oxh3R4qk79&open=AWcfCxK8d6oxh3R4qk79)

[ISSUE_KEY]: AWcfCxK8d6oxh3R4qk79